src/deepsparse/transformers/haystack/pipeline.py

```diff
@@ -0,0 +1,517 @@
+        retriever_kwargs["batch_size"] = batch_size
+        retriever_kwargs["num_cores"] = num_cores
+        retriever_kwargs["scheduler"] = scheduler
+        retriever_kwargs["input_shapes"] = input_shapes
+        retriever_kwargs["alias"] = alias
+        retriever_kwargs["context"] = context
+        retriever_kwargs["sequence_length"] = self._sequence_length
+        self._config = self._parse_config(config)
+
+        self.initialize_pipeline(retriever_kwargs)
+        if docs is not None:
+            self.write_documents(docs, overwrite=True)
+
+    def _merge_retriever_args(
+        self,
+        config_retriever_args: Dict[str, Any],
+        init_retriever_kwargs: Dict[str, Any],
+    ) -> Dict[str, Any]:
+        """
+        Merges retriever args given in config with args given at
+        HaystackPipeline initialization. Raises errors for conflicts
+
+        :param config_retriever_args: arguments given in config
+        :param init_retriever_kwargs: retriever arguments given at
+            HaystackPipeline initialization
+        :return: merged arguments from both inputs
+        """
+        # check for conflicting arguments
+        for key in init_retriever_kwargs.keys():
+            if key in config_retriever_args.keys():
+                raise ValueError(
+                    f"Found {key} in both HaystackPipeline arguments and config "
+                    "retriever_args. Specify only one"
+                )
+
+        # merge
+        merged_args = {}
+        merged_args.update(config_retriever_args)
+        merged_args.update(init_retriever_kwargs)
+
+        # rename pipeline arguments to fit retriever arguments
+        if "extraction_strategy" in merged_args:
+            if "pooling_strategy" in merged_args:
+                raise ValueError(
+                    "Found both extraction_strategy and pooling_strategy in "
+                    "arguments. Specify only one"
+                )
+            merged_args["pooling_strategy"] = merged_args["extraction_strategy"]
+            del merged_args["extraction_strategy"]
+        if "sequence_length" in merged_args:
+            if "max_seq_len" in merged_args:
+                raise ValueError(
+                    "Found both sequence_length and max_seq_len in "
+                    "arguments. Specify only one"
+                )
+            merged_args["max_seq_len"] = merged_args["sequence_length"]
+            del merged_args["sequence_length"]
+
+        return merged_args
+
+    def initialize_pipeline(self, init_retriever_kwargs: Dict[str, Any]) -> None:
+        """
+        Instantiate Haystack nodes needed to run pipeline
+
+        :param init_retriever_kwargs: retriever args passed at the initialization
+        of this pipeline
+        :return: None
+        """
+        # intialize document store from haystack
+        DocumentStoreModule = importlib.import_module(".document_stores", "haystack")
```

Comment:
let's move all these dynamic imports to the top of this file and rename the `*Module` part to `*Haystack`